core/src/main/java/org/infinispan/stream/impl/LockStreamImpl.java

```diff
@@ -0,0 +1,231 @@
+      } else {
+         return new LockStreamImpl<>(resultingStream, predicate);
+      }
+   }
+
+   @Override
+   public LockedStream<K, V> filter(final Predicate<? super CacheEntry<K, V>> predicate) {
+      Objects.nonNull(predicate);
+      Predicate<? super CacheEntry<K, V>> usedPredicate;
+      if (this.predicate != null) {
+         usedPredicate = (SerializablePredicate<? super CacheEntry<K, V>>) e -> this.predicate.test(e) && predicate.test(e);
+      } else {
+         usedPredicate = predicate;
+      }
+      return new LockStreamImpl<>(realStream, usedPredicate);
+   }
+
+   @Override
+   public void forEach(BiConsumer<Cache<K, V>, ? super CacheEntry<K, V>> biConsumer) {
+      realStream.forEach(new CacheEntryConsumer<>(biConsumer, predicate));
+   }
+
+   @Override
+   public BaseCacheStream sequentialDistribution() {
+      return newOrReuse(realStream.sequentialDistribution());
+   }
+
+   @Override
+   public BaseCacheStream parallelDistribution() {
+      return newOrReuse(realStream.parallelDistribution());
+   }
+
+   @Override
+   public BaseCacheStream filterKeySegments(Set<Integer> segments) {
+      return newOrReuse(realStream.filterKeySegments(segments));
+   }
+
+   @Override
+   public BaseCacheStream filterKeys(Set<?> keys) {
+      return newOrReuse(realStream.filterKeys(keys));
+   }
+
+   @Override
+   public BaseCacheStream distributedBatchSize(int batchSize) {
+      return newOrReuse(realStream.distributedBatchSize(batchSize));
+   }
+
+   @Override
+   public BaseCacheStream segmentCompletionListener(SegmentCompletionListener listener) {
+      throw new UnsupportedOperationException("LockStream doesn't support completion listener");
+   }
+
+   @Override
+   public BaseCacheStream disableRehashAware() {
+      return newOrReuse(realStream.disableRehashAware());
+   }
+
+   @Override
+   public BaseCacheStream timeout(long timeout, TimeUnit unit) {
+      return newOrReuse(realStream.timeout(timeout, unit));
+   }
+
+   @Override
+   public Iterator<CacheEntry<K, V>> iterator() {
+      return realStream.iterator();
+   }
+
+   @Override
+   public Spliterator<CacheEntry<K, V>> spliterator() {
+      return realStream.spliterator();
+   }
+
+   @Override
+   public boolean isParallel() {
+      return realStream.isParallel();
+   }
+
+   @Override
+   public LockedStream<K, V> sequential() {
+      return newOrReuse(realStream.sequential());
+   }
+
+   @Override
+   public LockedStream<K, V> parallel() {
+      return newOrReuse(realStream.parallel());
+   }
+
+   @Override
+   public LockedStream<K, V> unordered() {
+      // This stream is always unordered
+      return this;
+   }
+
+   @Override
+   public LockedStream<K, V> onClose(Runnable closeHandler) {
+      return newOrReuse(realStream.onClose(closeHandler));
+   }
+
+   @Override
+   public void close() {
+      realStream.close();
+   }
+
+   @SerializeWith(value = CacheEntryConsumer.CacheEntryConsumerExternalizer.class)
```

Comment:
The name actually doesn't matter much in this case. This is using the less performant Serializer which actually ships across the entire Serializer class as is. If we wanted this to be as highest performant we would want to extract an actual `AdvancedExternalizer` id. I figured this operation is not about # of invocations as it has to perform the `BiConsumer` for every entry after obtaining the key. This requires a lot less code and doesn't require a reserved ID. Do you think I should also convert this over? It just seemed like overkill.